core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -282,45 +349,49 @@ private List<String> getNames( IMetaStore metaStore, ConnectionProvider<? extend
    * @param provider A provider
    * @return A list of named connection names
    */
-  private List<String> getNames( ConnectionProvider<? extends ConnectionDetails> provider ) {
+  private List<String> loadNames( ConnectionProvider<? extends ConnectionDetails> provider ) {
     if ( metaStoreSupplier == null || metaStoreSupplier.get() == null ) {
       return Collections.emptyList();
     }
     return getNames( metaStoreSupplier.get(), provider );
   }
 
+  /**
+   * Get the names of named connections by provider from default meta store
+   *
+   * @param provider A provider
+   * @return A list of named connection names
+   */
+  private List<String> getNames( ConnectionProvider<? extends ConnectionDetails> provider ) {
+    initialize();
+    return namesByConnectionProvider.get( provider.getName() );
+  }
+
   /**
    * Get the names of named connections by provider from specified meta store
    *
    * @param metaStore  A meta store
-   * @param clearCache Whether or not to clear cache
+   * @param clearCache Whether or not to clear cache. (UNUSED)
    * @return A list of named connection names
    */
-  public List<String> getNames( IMetaStore metaStore, boolean clearCache ) {
-    if ( clearCache ) {
-      nameCache.clear();
-    }
-    if ( !nameCache.isEmpty() ) {
-      return nameCache;
-    }
+  private List<String> getNames( IMetaStore metaStore, boolean clearCache ) {
     List<String> detailNames = new ArrayList<>();
     List<ConnectionProvider<? extends ConnectionDetails>> providers =
       Collections.list( connectionProviders.elements() );
     for ( ConnectionProvider<? extends ConnectionDetails> provider : providers ) {
       detailNames.addAll( getNames( metaStore, provider ) );
     }
-    nameCache.addAll( detailNames );
     return detailNames;
   }
 
   /**
    * Get the names of named connections by provider from the default meta store
    *
-   * @param clearCache - Whether or not to clear cache
+   * @param clearCache - Whether or not to clear cache (unused)
    * @return A list of named connection names
    */
   public List<String> getNames( boolean clearCache ) {
```

Comment:
Same comment as above. clearCache parameter is redundant

Reply:
same.